File: php/src/CourtListener/Api/DisclosurePositions.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * DisclosurePositions API endpoint
 */
class DisclosurePositions extends BaseApi
{
    protected string $endpoint = 'disclosure-positions/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List disclosure positions with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listDisclosurePositions(array $params = []): array
    {
        return $this->list($params);
    }

    /**
     * Get a specific disclosure positions by ID
     *
     * @param int|string $id DisclosurePositions ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getDisclosurePositions($id, array $params = []): array
    {
        return $this->get($id, $params);
    }

    /**
     * Search disclosure positions
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchDisclosurePositions(array $params = []): array
    {
        return $this->search($params);
    }
}

File: python/courtlistener/api/disclosure_positions.py
"""
Disclosure Positions API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class DisclosurePositionsAPI(BaseAPI):
    """API for accessing disclosure position data from financial disclosures."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "disclosure-positions/"
    
    def list(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List disclosure position records.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of disclosure positions
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, position_id: int) -> Dict[str, Any]:
        """
        Get a specific disclosure position by ID.
        
        Args:
            position_id: Disclosure position ID
        
        Returns:
            Disclosure position data
        """
        return self.client.get(f"{self.endpoint}{position_id}/")
    
    def paginate(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated disclosure positions.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for disclosure positions
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)
